File: common/src/main/java/org/bread_experts_group/breadlib/test/client/TasksClientTest.java
package org.bread_experts_group.breadlib.test.client;

import net.minecraft.client.Minecraft;
import org.bread_experts_group.breadlib.BreadLib;
import org.bread_experts_group.breadlib.task.TaskManager;
import org.bread_experts_group.breadlib.task.network.NetworkTask;
import org.bread_experts_group.breadlib.task.render.LayeredDrawTask;
import org.bread_experts_group.breadlib.test.ClientboundPacketTest;
import org.bread_experts_group.breadlib.util.Color;

public class TasksClientTest {
	public static void renderTest() {
//		TaskManager.newTask(LevelRenderTask.class, task -> BreadLib.LOGGER.info(task.stage));
	}

	public static void layeredDrawTest() {
		TaskManager.newTask(LayeredDrawTask.class, task -> task.add(BreadLib.modLoc("layered_draw", "test_layer"), ((guiGraphics, deltaTracker) -> {
//				Window window = Minecraft.getInstance().getWindow();
			guiGraphics.fill(0, 0, 25, 12, Color.BLACK);
			guiGraphics.drawString(Minecraft.getInstance().font, "Breadlib", 1, 1, Color.WHITE, false);
		})));
	}

	public static void networkTest() {
		TaskManager.newTask(NetworkTask.class, task -> task.addClientbound(
				ClientboundPacketTest.class,
				ClientboundPacketTest.TYPE,
				ClientboundPacketTest.STREAM_CODEC,
				ClientboundPacketTest::handleClientbound
		));
	}
}

File: common/src/main/kotlin/org/bread_experts_group/breadlib/test/client/TasksClientTest.kt
package org.bread_experts_group.breadlib.test.client

import net.minecraft.client.DeltaTracker
import net.minecraft.client.Minecraft
import net.minecraft.client.gui.GuiGraphics
import net.minecraft.network.chat.Component
import org.bread_experts_group.breadlib.BreadLib
import org.bread_experts_group.breadlib.BreadLib.modLoc
import org.bread_experts_group.breadlib.platform.PlatformServices
import org.bread_experts_group.breadlib.task.TaskManager.newTask
import org.bread_experts_group.breadlib.task.render.LayeredDrawTask
import org.bread_experts_group.breadlib.util.Color
import org.bread_experts_group.breadlib.util.minecraft

object TasksClientTest {
	fun renderTest() {
//		TaskManager.newTask(LevelRenderTask.class, task -> BreadLib.LOGGER.info(task.stage));
	}

	fun layeredDrawTest() {
		newTask { task: LayeredDrawTask ->
			val platform = PlatformServices.PLATFORM.platformName
			val debugInfo = Component.literal("${BreadLib.MOD_ID} v${BreadLib.MOD_VERSION} | $platform")
			val width = Minecraft.getInstance().font.width(debugInfo)
			task.add(modLoc("layered_draw", "test_layer")) { guiGraphics: GuiGraphics, _: DeltaTracker ->
				guiGraphics.fill(0, 0, width, 12, Color.BLACK)
				guiGraphics.drawString(Minecraft.getInstance().font, debugInfo, 2, 2, Color.ORANGE, false)
			}

			task.add(BreadLib.modLoc("dim_overlay")) { guiGraphics, deltaTracker ->
				val dimension = minecraft!!.level!!.dimension()
				guiGraphics.drawString(minecraft!!.font, "current dim: ${dimension.location()}", 0, 15, Color.WHITE)
			}
		}
	}
}
